lib/artifact-search.js

```diff
@@ -68,29 +78,56 @@ module.exports = (bluebird, error, logger, requestOptions, requestPromise, respo
 
 
         /**
+         * @typedef {Object} searchParameters
+         * @property {(string|string[])} [search] A string or array of strings
+         *                                        for the search criteria. Can
+         *                                        be undefined (optional).
+         * @property {(string|string[])} [sort] A string or array of strings
+         *                                      for how the results will be
+         *                                      sorted. Can be undefined
+         *                                      (optional).
+         * @property {number} [limit] A number denoting how many results should
+         *                            be returned from the search. Can be
+         *                            undefined (optional).
+         */
+
+
+        /**
          * Sends and returns a request with the provided search criteria.
          *
-         * @param {(string|string[])} search A string or array of strings for
-         *                                   the search criteria.
-         * @param {(string|string[])} sort A string or array of strings for
-         *                                 how the results will be sorted.
+         * @param {searchParameters} searchParameters
          * @return {Promise.<Object>}
          */
-        search(search, sort) {
+        search(searchParameters) {
             var options, urls;
 
             logger.debug(`In ${this.constructor.name}.search()`);
+            logger.debug(`Search parameters: ${JSON.stringify(searchParameters, null, 4)}`);
 
-            return validateSearch(search).then(() => {
+            if (!searchParameters) {
+                searchParameters = {};
+            }
+
+            return validateSearch(searchParameters.search).then(() => {
                 options = requestOptions.createOptions(this.authToken);
                 options.json = true;
                 options.url = new URI(this.host)
                         .path(URI.joinPaths(`${this.refName}/artifact/`, `${this.path}/_search`))
                         .toString();
-                options.body = {
-                    search,
-                    sort
-                };
+                options.body = {};
+
+                if (searchParameters.search) {
+                    options.body.search = searchParameters.search;
+                }
+
+                if (searchParameters.sort) {
+                    options.body.sort = searchParameters.sort;
+                }
+
+                if (searchParameters.limit) {
+                    options.body.limit = searchParameters.limit;
+                }
+
```

Comment:
You can just set the searchParameters as the body, right? `options.body = searchParameters` that is essentially what you are doing already. When it is loaded as JSON any undefined properties will just be omitted.